imports/wasi_snapshot_preview1/fs_test.go

```diff
@@ -1237,7 +1331,10 @@ func Test_fdReaddir(t *testing.T) {
 			require.True(t, ok)
 
 			if tc.expectedMem != nil {
-				require.Equal(t, tc.expectedMem, mem)
+				if tc.expectedMemSize == 0 {
```

Comment:
I had to add/change this for the test case "read second and a little more", that results in a bufused of the complete buffer because it fits one item but not another header but since the last bytes of the buffer could be anything because it's not zero'ed out I had to come up with a way to exclude these last bytes from the comparison.